extlinks/links/admin.py

```diff
@@ -19,6 +27,12 @@ class LinkSearchTotalAdmin(admin.ModelAdmin):
 
 class LinkEventAdmin(admin.ModelAdmin):
     list_display = ("link", "timestamp", "domain", "username", "change")
+    readonly_fields = ["url_pattern_display"]
+    exclude = ["content_type", "object_id"]
+
```

Comment:
```suggestion
list_select_related = ["username", "content_type"]
```
Let's add this to improve performance.